lib/msf/base/sessions/command_shell.rb

```diff
@@ -312,18 +312,18 @@ def cmd_shell(*args)
   end
 
   #
-  # Check if there is a binary in PATH env
+  # Returns path of a binary in PATH env.
   #
   def binary_exists(binary)
     print_status("Trying to find binary(#{binary}) on target machine")
-    binary_path = shell_command_token("which #{binary}").to_s.strip
-    if binary_path.eql?("#{binary} not found") || binary_path == ""
-      print_error(binary_path)
+    binary_path = shell_command_token("command -v '#{binary}' && echo true")
+    if binary_path.eql?("")
```

Comment:
The results of `shell_command_token` can be `nil`.

```suggestion
    if binary_path.to_s.eql?("")
```